Mathlib/Analysis/LegendrePoly.lean

```diff
@@ -0,0 +1,138 @@
+/-
+Copyright (c) 2024 Junqi Liu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Junqi Liu
+-/
+import Mathlib.Algebra.Polynomial.Basic
+import Mathlib.Algebra.BigOperators.Ring
+import Mathlib.Algebra.Order.BigOperators.Group.Finset
+import Mathlib.Algebra.Polynomial.AlgebraMap
+import Mathlib.Algebra.Polynomial.Derivative
+import Mathlib.Analysis.Normed.Module.Basic
+import Mathlib.Data.Nat.Choose.Sum
+import Mathlib.Data.Real.Basic
+import Mathlib.Tactic.Linarith
+import Mathlib.Tactic.Ring
+
+/-!
+# Legendre Polynomials
+
+In this file, we define the Legendre polynomials `Legendre n` for `n : ℕ` as a polynomial in `ℝ[X]`.
+We prove some basic properties of the Legendre polynomials.
+
+-/
+
+open scoped Nat
+open BigOperators Finset
+
+variable {R : Type*}
+
+namespace Polynomial
+
+/--  `Legendre n` is the polynomial defined in terms of derivatives of order n.  -/
+noncomputable def Legendre (n : ℕ) : ℝ[X] :=
+  C (n ! : ℝ)⁻¹ * derivative^[n] (X ^ n * (1 - X) ^ n)
+
+lemma Finsum_iterate_deriv [CommRing R] (k : ℕ) (h : ℕ → ℕ) :
```

Comment:
This lemma does not seem to be specific to Legendre polynomials, right? Please call `#find_home! <lemma_name>` below it, to find a good place.

Note that `Finsum` is a different concept from `Finset.sum`. So the name should also be updated.